modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm

```diff
@@ -297,49 +427,119 @@
 * If the player's penis is near or at the maximum size and they're wearing clothing, it presses against the player's clothes and causes brute damage.
-* Overdosing makes you male if female and shrinks the player's breasts to a minimum size.
+* If gender change preference is enabled: Overdosing makes you male if female, makes the player grow a cock, and shrinks the player's breasts to a minimum size.
+* If the gender change preference is not enabled: Overdosing will just make you grow a cock if you don't have one.
 */
 
 /datum/reagent/drug/aphrodisiac/penis_enlarger
-	name = "Incubus draft"
+	name = "incubus draft"
 	description = "A volatile collodial mixture derived from various masculine solutions that encourages a larger gentleman's package via a potent testosterone mix."
 	color = "#888888"
 	taste_description = "chinese dragon powder"
-	overdose_threshold = 17 //ODing makes you male and removes female genitals
-	metabolization_rate = 0.5
+	overdose_threshold = 20 ///ODing makes you male and shrinks female genitals if gender change prefs are enabled. Otherwise, grows a cock.
+	metabolization_rate = 0.25
 	life_pref_datum = /datum/preference/toggle/erp/penis_enlargement
-	overdose_pref_datum = /datum/preference/toggle/erp/gender_change
+	overdose_pref_datum = /datum/preference/toggle/erp/penis_enlargement ///Changed from gender_change in order to have that as a separate feature within overdose.
 
 /datum/reagent/drug/aphrodisiac/penis_enlarger/life_effects(mob/living/carbon/human/exposed_mob)
 	var/obj/item/organ/genital/penis/mob_penis = exposed_mob.getorganslot(ORGAN_SLOT_PENIS)
 	enlargement_amount += enlarger_increase_step
+	/// Add yet another check because I hate errors!!
+	if(!mob_penis)
+		return
 	if(enlargement_amount >= enlargement_threshold)
-		if(mob_penis?.genital_size >= penis_max_length || mob_penis?.girth >= penis_max_girth)
+		if(mob_penis?.genital_size >= penis_max_length)
 			return ..()
 		mob_penis.genital_size += penis_length_increase_step
-		if(prob(20))
-			mob_penis.girth += penis_girth_increase_step
+		///Improvision to girth to not make it random chance.
+		if(mob_penis?.girth < penis_max_girth) ///Because any higher is ridiculous. However, should still allow for regular penis growth.
+			mob_penis.girth = round(mob_penis.girth + (mob_penis.genital_size/mob_penis.girth))
 		mob_penis.update_sprite_suffix()
 		exposed_mob.update_body()
 		enlargement_amount = 0
-
-	if(ISINRANGE_EX(mob_penis?.genital_size, (penis_max_length - 2), penis_max_length) && (exposed_mob.w_uniform || exposed_mob.wear_suit))
+		/// Words for the cock when huge.
+		var/static/list/words_for_bigger_cock = list("huge", "massive", "gigantic", "rather lengthy", "colossal", "hefty")
+		/// Synonyms for cock.
+		var/static/list/cock_text_list = list("cock", "penis", "dick", "member")
+		/// Synonyms for bigger cock.
+		var/static/list/bigger_cock_text_list = list("rod", "shaft", "cock", "penis", "dick", "member")
```

Comment:
```suggestion
		var/static/list/bigger_cock_text_list = list("rod", "shaft", "cock", "penis", "dick", "member", "richard", "johnston", "johnson",)
```
right and PROPER

Reply:
also trailing commas, add commas to the end of all em :)

Reply:
wait where does johnston come from

Reply:
the short hand of richard is dick

Reply:
fair